pkg/cmd/inspect.go

```diff
@@ -599,7 +573,28 @@ func (o *InspectOptions) gatherPodData(destDir, namespace string, pod *corev1.Po
 	return nil
 }
 
-func (o *InspectOptions) gatherContainerData(destDir string, pod *corev1.Pod, container *corev1.Container, metricsPort *util.RemoteContainerPort) error {
+func (o *InspectOptions) gatherContainerInfo(destDir string, pod *corev1.Pod, container corev1.Container) error {
+	if err := o.gatherContainerAllLogs(path.Join(destDir, "/"+container.Name), pod, &container); err != nil {
+		return err
+	}
+
+	if len(container.Ports) == 0 {
```

Comment:
Also, a few operator pods, for example, don't have anything under `container.Ports` as it currently stands

Reply:
An immediate example I can think of is the `openshift-kube-apiserver-operator` pod